rdmd.d

```diff
@@ -897,9 +897,9 @@ string which(string path)
     if (path.canFind(dirSeparator) || altDirSeparator != "" && path.canFind(altDirSeparator)) return path;
     string[] extensions = [""];
     version(Windows) extensions ~= environment["PATHEXT"].split(pathSeparator);
-    foreach (extension; extensions)
+    foreach (envPath; environment["PATH"].splitter(pathSeparator))
     {
-        foreach (envPath; environment["PATH"].splitter(pathSeparator))
+        foreach (extension; extensions)
```

Comment:
FWIW `dub` just assumes this to be `.exe`: https://github.com/dlang/dub/blob/350a0315c38fab9d3d0c4c9d30ff6bb90efb54d6/source/dub/dub.d#L1183

Reply:
Actually since we are using `spawnProcess` instead of `spawnShell`, it only works with `.exe` :)  I actaully do have a `dmd.bat` and `rdmd.bat` that I use on my windows machines.  I usually have a single directory in my path that I add tools to, I find it easier than managing paths in the PATH environment variable.